src/main/webapp/app/exercise/result-history/result-history.component.html

```diff
@@ -10,19 +10,19 @@
                     [ngClass]="
                         result.id === selectedResultId()
                             ? 'text-primary'
-                            : getTextColorClass(result, evaluateTemplateStatus(exercise(), result.submission?.participation, result, false, MissingResultInfo.NONE))
+                            : getTextColorClass(result, participation()!, evaluateTemplateStatus(exercise(), participation(), result, false, MissingResultInfo.NONE))
```

Comment:
Either define participation directly as `Participation` that cannot be undefined or add an undefined check before, just asserting that it is not undefined is not a good and maintainable way to program this in my opinion